packages/builder/src/lib/script-checkpoints.ts

```diff
@@ -1,105 +1,310 @@
 
+import deepEqual from "deep-equal";
 import * as fs from "fs";
 import path from "path";
 import YAML from "yaml";
 
-import { AlgobDeployer, AlgobRuntimeEnv, ScriptCheckpoint, ScriptNetCheckpoint } from "../types";
+import { BuilderError } from "../internal/core/errors";
+import { ERRORS } from "../internal/core/errors-list";
+import {
+  AccountDef,
+  AlgobDeployer,
+  AlgobRuntimeEnv,
+  ASAInfo,
+  ASCInfo,
+  CheckpointData,
+  ScriptCheckpoints,
+  ScriptNetCheckpoint
+} from "../types";
 
 export const scriptsDirectory = "scripts";
 const artifactsPath = "artifacts";
+const checkpointFileSuffix = ".cp.yaml";
 
 export function toCheckpointFileName (scriptName: string): string {
-  return path.join(artifactsPath, scriptName + ".cp.yaml");
+  return path.join(artifactsPath, scriptName + checkpointFileSuffix);
 }
 
-export function appendToCheckpoint (
-  prev: ScriptCheckpoint,
-  networkName: string,
-  append: ScriptNetCheckpoint
-): ScriptCheckpoint {
-  if (prev[networkName]) {
-    prev[networkName].timestamp = append.timestamp;
-    prev[networkName].metadata = Object.assign({}, prev[networkName].metadata, append.metadata);
-    return prev;
-  }
-  prev[networkName] = append;
-  return prev;
+export function registerASA (
+  cp: ScriptNetCheckpoint, name: string, creator: string): ScriptNetCheckpoint {
+  cp.asa[name] = { creator: creator };
+  return cp;
+}
+
+export function registerASC (
+  cp: ScriptNetCheckpoint, name: string, creator: string): ScriptNetCheckpoint {
+  cp.asc[name] = { creator: creator };
+  return cp;
 }
```

Comment:
This should be a method on ScriptNetCheckpoint

Reply:
Previous comment about `assert.deepEquals`